src/awala.spec.ts

```diff
@@ -34,206 +44,243 @@ describe('awala', () => {
-      expect(dbResult!.awalaPda.toString('base64')).toBe(AWALA_PDA);
-      expect(dbResult!.signature.toString('base64')).toBe(SIGNATURE);
-      expect(dbResult!.memberBundleStartDate).toBeDate();
-      expect(dbResult!.memberBundleStartDate.toISOString()).toBe(futureTimestamp);
-      expect(mockLogging.logs).toContainEqual(
-        partialPinoLog('info', 'Member bundle request created', {
-          publicKeyId: MEMBER_PUBLIC_KEY_MONGO_ID,
-        }),
-      );
+    await createMemberBundleRequest(methodInput, serviceOptions);
+    const dbResult = await memberBundleRequestModel.findOne({
+      publicKeyId: memberPublicKey._id.toString(),
     });
+    expect(dbResult).not.toBeNull();
+    expect(dbResult!.memberId).toBe(MEMBER_MONGO_ID);
+    expect(dbResult!.awalaPda.toString('base64')).toBe(AWALA_PDA);
+    expect(dbResult!.signature.toString('base64')).toBe(SIGNATURE);
+    expect(dbResult!.memberBundleStartDate).toBeDate();
+    expect(dbResult!.memberBundleStartDate.toISOString()).toBe(futureTimestamp);
+  });
 
-    test('Member bundle data should be updated', async () => {
-      const data = {
-        publicKeyId: MEMBER_PUBLIC_KEY_MONGO_ID,
-        memberBundleStartDate: new Date(),
-        signature: 'test',
-        awalaPda: 'test',
-      };
-      await memberBundleRequestModel.create(data);
-
-      await createMemberBundleRequest(methodInput, serviceOptions);
-      const dbResult = await memberBundleRequestModel.findOne({
-        publicKeyId: MEMBER_PUBLIC_KEY_MONGO_ID,
-      });
-      expect(dbResult).not.toBeNull();
-      expect(dbResult!.awalaPda.toString('base64')).toBe(AWALA_PDA);
-      expect(dbResult!.signature.toString('base64')).toBe(SIGNATURE);
-      expect(dbResult!.memberBundleStartDate).toBeDate();
-      expect(dbResult!.memberBundleStartDate.toISOString()).toBe(futureTimestamp);
+  test('Existing data should not create new entry', async () => {
+    await memberBundleRequestModel.create({
+      ...methodInput,
+      memberId: MEMBER_MONGO_ID,
     });
 
-    test('Existing data should not create new entry', async () => {
-      await memberBundleRequestModel.create(methodInput);
+    await createMemberBundleRequest(methodInput, serviceOptions);
 
-      await createMemberBundleRequest(methodInput, serviceOptions);
+    const countResult = await memberBundleRequestModel.count({
+      publicKeyId: memberPublicKey._id.toString(),
+    });
+    expect(countResult).toBe(1);
+  });
 
-      const countResult = await memberBundleRequestModel.count({
+  test('Invalid public key id should be refused', async () => {
+    const result = await createMemberBundleRequest(
+      {
+        ...methodInput,
         publicKeyId: MEMBER_PUBLIC_KEY_MONGO_ID,
-      });
-      expect(countResult).toBe(1);
+      },
+      serviceOptions,
+    );
+
+    expect(result.didSucceed).not.toBeTrue();
+  });
+
+  test('Invalid public key Id should not create member bundle request', async () => {
```

Comment:
Looking at the other tests in this file, wouldn't it be more consistent to bundle this together with the test above? See:

https://github.com/relaycorp/veraid-authority/blob/291f81f7bbf8a4265341290a8e800642aba5a38c/src/awala.spec.ts#L52